src/lib/kit/components/Inputs/TimeRangeSelector/TimeRangeSelector.tsx

```diff
@@ -0,0 +1,107 @@
+import React from 'react';
+
+import cloneDeep from 'lodash/cloneDeep';
+import isString from 'lodash/isString';
+import set from 'lodash/set';
+
+import {FieldValue, ObjectIndependentInput, ValidateError, isStringSpec} from '../../../../core';
+import {END_TIME, START_TIME} from '../../../constants/common';
+
+import {TimeRangeSelect} from './components';
+import {filterTimeArray} from './utils';
+
+export const TimeRangeSelector: ObjectIndependentInput = (props) => {
+    const {spec, input, name, Layout} = props;
+
+    const {startTimeSpec, endTimeSpec} = React.useMemo(() => {
+        const [startTimeSpec, endTimeSpec] = [START_TIME, END_TIME].map((key) => {
+            const property = spec.properties?.[key];
+
+            if (isStringSpec(property)) {
+                const _spec = cloneDeep(property);
```

Comment:
you have no reason to clone this spec

it looks like this memo might be like this

```
const [startTimeSpec, endTimeSpec] = React.useMemo(() =>
   [START_TIME, END_TIME].map(
      (key) => isStringSpec(spec.properties?.[key]) ? spec.properties?.[key] : undefined)
, [spec.properties]);
```